src/main/java/com/google/api/codegen/util/SymbolTable.java

```diff
@@ -89,4 +114,24 @@ private String getAndSaveSuffix(String desiredName) {
     symbolTable.add(desiredName + Integer.toString(i));
     return Integer.toString(i);
   }
+
+  /**
+   * Returns the next underscore suffix that makes desiredName unique.
+   *
+   * <p>Stores the joined desiredName/suffix in an internal map. Assumes that desiredName is already
+   * in the symbolTable.
+   *
+   * <p>For example, if "foo" is passed, "_" is returned. If "foo" is passed again, "__" is
```

Comment:
As in my comments https://github.com/googleapis/toolkit/issues/564#issuecomment-257925186, a current negative outcome of the numeric scheme is that in some instances, we get variable `foo` used to iterate over elements of `foo2`, which is quite confusing. Underscores are not great in this instance either, but at least they don't contradict the actual relationship so blatantly. Perhaps this case would be better resolved by handling collisions with iterators explicitly, so that `foo` can be used to iterate over `foo_page` or whatever a more appropriate name might be for the collection.